### PR TITLE
[gateway] support descriptions in gateway list items

### PR DESCRIPTION
**Pull request by rexscaria**

### CODE REVIEW

File: internal/sdkv2provider/schema_cloudflare_teams_list.go [hunk 1]
```diff
@@ -34,10 +34,34 @@ func resourceCloudflareTeamsListSchema() map[string]*schema.Schema {
 		"items": {
 			Type:        schema.TypeSet,
 			Optional:    true,
+			MaxItems:    1,
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_teams_list.go:37]: this breaks current tests; is that expected?

**Comment by jacobbednarz** [comment]
this breaks the existing test suite; is that expected?

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsList_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareTeamsList_Basic
    resource_cloudflare_teams_list_test.go:28: Step 1/1 error: Error running pre-apply plan: exit status 1
        
        Error: Too many list items
        
          with cloudflare_teams_list.yzosksvbqf,
          on terraform_plugin_test.tf line 17, in resource "cloudflare_teams_list" "yzosksvbqf":
          17: 	items       = ["asdf-1234", "asdf-5678"]
        
        Attribute items supports 1 item maximum, but config has 2 declared.
--- FAIL: TestAccCloudflareTeamsList_Basic (1.01s)
=== RUN   TestAccCloudflareTeamsList_BasicWithDescription
--- PASS: TestAccCloudflareTeamsList_BasicWithDescription (3.25s)
=== RUN   TestAccCloudflareTeamsList_LottaListItems
    resource_cloudflare_teams_list_test.go:98: Step 1/1 error: Error running pre-apply plan: exit status 1
        
        Error: Too many list items
        
          with cloudflare_teams_list.nbjktepvev,
          on terraform_plugin_test.tf line 17, in resource "cloudflare_teams_list" "nbjktepvev":
          17: 	items       = ["example-0","example-1","example-2","example-3","example-4","example-5","example-6","example-7","example-8","example-9","example-10","example-11","example-12","example-13","example-14","example-15","example-16","example-17","example-18","example-19","example-20","example-21","example-22","example-23","example-24","example-25","example-26","example-27","example-28","example-29","example-30","example-31","example-32","example-33","example-34","example-35","example-36","example-37","example-38","example-39","example-40","example-41","example-42","example-43","example-44","example-45","example-46","example-47","example-48","example-49","example-50","example-51","example-52","example-53","example-54","example-55","example-56","example-57","example-58","example-59","example-60","example-61","example-62","example-63","example-64","example-65","example-66","example-67","example-68","example-69","example-70","example-71","example-72","example-73","example-74","example-75","example-76","example-77","example-78","example-79","example-80","example-81","example-82","example-83","example-84","example-85","example-86","example-87","example-88","example-89","example-90","example-91","example-92","example-93","example-94","example-95","example-96","example-97","example-98","example-99","example-100","example-101","example-102","example-103","example-104","example-105","example-106","example-107","example-108","example-109","example-110","example-111","example-112","example-113","example-114","example-115","example-116","example-117","example-118","example-119","example-120","example-121","example-122","example-123","example-124","example-125","example-126","example-127","example-128","example-129","example-130","example-131","example-132","example-133","example-134","example-135","example-136","example-137","example-138","example-139","example-140","example-141","example-142","example-143","example-144","example-145","example-146","example-147","example-148","example-149","example-150","example-151","example-152","example-153","example-154","example-155","example-156","example-157","example-158","example-159","example-160","example-161","example-162","example-163","example-164","example-165","example-166","example-167","example-168","example-169","example-170","example-171","example-172","example-173","example-174","example-175","example-176","example-177","example-178","example-179","example-180","example-181","example-182","example-183","example-184","example-185","example-186","example-187","example-188","example-189","example-190","example-191","example-192","example-193","example-194","example-195","example-196","example-197","example-198","example-199","example-200","example-201","example-202","example-203","example-204","example-205","example-206","example-207","example-208","example-209","example-210","example-211","example-212","example-213","example-214","example-215","example-216","example-217","example-218","example-219","example-220","example-221","example-222","example-223","example-224","example-225","example-226","example-227","example-228","example-229","example-230","example-231","example-232","example-233","example-234","example-235","example-236","example-237","example-238","example-239","example-240","example-241","example-242","example-243","example-244","example-245","example-246","example-247","example-248","example-249","example-250","example-251","example-252","example-253","example-254","example-255","example-256","example-257","example-258","example-259","example-260","example-261","example-262","example-263","example-264","example-265","example-266","example-267","example-268","example-269","example-270","example-271","example-272","example-273","example-274","example-275","example-276","example-277","example-278","example-279","example-280","example-281","example-282","example-283","example-284","example-285","example-286","example-287","example-288","example-289","example-290","example-291","example-292","example-293","example-294","example-295","example-296","example-297","example-298","example-299","example-300","example-301","example-302","example-303","example-304","example-305","example-306","example-307","example-308","example-309","example-310","example-311","example-312","example-313","example-314","example-315","example-316","example-317","example-318","example-319","example-320","example-321","example-322","example-323","example-324","example-325","example-326","example-327","example-328","example-329","example-330","example-331","example-332","example-333","example-334","example-335","example-336","example-337","example-338","example-339","example-340","example-341","example-342","example-343","example-344","example-345","example-346","example-347","example-348","example-349","example-350","example-351","example-352","example-353","example-354","example-355","example-356","example-357","example-358","example-359","example-360","example-361","example-362","example-363","example-364","example-365","example-366","example-367","example-368","example-369","example-370","example-371","example-372","example-373","example-374","example-375","example-376","example-377","example-378","example-379","example-380","example-381","example-382","example-383","example-384","example-385","example-386","example-387","example-388","example-389","example-390","example-391","example-392","example-393","example-394","example-395","example-396","example-397","example-398","example-399","example-400","example-401","example-402","example-403","example-404","example-405","example-406","example-407","example-408","example-409","example-410","example-411","example-412","example-413","example-414","example-415","example-416","example-417","example-418","example-419","example-420","example-421","example-422","example-423","example-424","example-425","example-426","example-427","example-428","example-429","example-430","example-431","example-432","example-433","example-434","example-435","example-436","example-437","example-438","example-439","example-440","example-441","example-442","example-443","example-444","example-445","example-446","example-447","example-448","example-449","example-450","example-451","example-452","example-453","example-454","example-455","example-456","example-457","example-458","example-459","example-460","example-461","example-462","example-463","example-464","example-465","example-466","example-467","example-468","example-469","example-470","example-471","example-472","example-473","example-474","example-475","example-476","example-477","example-478","example-479","example-480","example-481","example-482","example-483","example-484","example-485","example-486","example-487","example-488","example-489","example-490","example-491","example-492","example-493","example-494","example-495","example-496","example-497","example-498","example-499","example-500","example-501","example-502","example-503","example-504","example-505","example-506","example-507","example-508","example-509","example-510","example-511","example-512","example-513","example-514","example-515","example-516","example-517","example-518","example-519","example-520","example-521","example-522","example-523","example-524","example-525","example-526","example-527","example-528","example-529","example-530","example-531","example-532","example-533","example-534","example-535","example-536","example-537","example-538","example-539","example-540","example-541","example-542","example-543","example-544","example-545","example-546","example-547","example-548","example-549","example-550","example-551","example-552","example-553","example-554","example-555","example-556","example-557","example-558","example-559","example-560","example-561","example-562","example-563","example-564","example-565","example-566","example-567","example-568","example-569","example-570","example-571","example-572","example-573","example-574","example-575","example-576","example-577","example-578","example-579","example-580","example-581","example-582","example-583","example-584","example-585","example-586","example-587","example-588","example-589","example-590","example-591","example-592","example-593","example-594","example-595","example-596","example-597","example-598","example-599","example-600","example-601","example-602","example-603","example-604","example-605","example-606","example-607","example-608","example-609","example-610","example-611","example-612","example-613","example-614","example-615","example-616","example-617","example-618","example-619","example-620","example-621","example-622","example-623","example-624","example-625","example-626","example-627","example-628","example-629","example-630","example-631","example-632","example-633","example-634","example-635","example-636","example-637","example-638","example-639","example-640","example-641","example-642","example-643","example-644","example-645","example-646","example-647","example-648","example-649","example-650","example-651","example-652","example-653","example-654","example-655","example-656","example-657","example-658","example-659","example-660","example-661","example-662","example-663","example-664","example-665","example-666","example-667","example-668","example-669","example-670","example-671","example-672","example-673","example-674","example-675","example-676","example-677","example-678","example-679","example-680","example-681","example-682","example-683","example-684","example-685","example-686","example-687","example-688","example-689","example-690","example-691","example-692","example-693","example-694","example-695","example-696","example-697","example-698","example-699","example-700","example-701","example-702","example-703","example-704","example-705","example-706","example-707","example-708","example-709","example-710","example-711","example-712","example-713","example-714","example-715","example-716","example-717","example-718","example-719","example-720","example-721","example-722","example-723","example-724","example-725","example-726","example-727","example-728","example-729","example-730","example-731","example-732","example-733","example-734","example-735","example-736","example-737","example-738","example-739","example-740","example-741","example-742","example-743","example-744","example-745","example-746","example-747","example-748","example-749","example-750","example-751","example-752","example-753","example-754","example-755","example-756","example-757","example-758","example-759","example-760","example-761","example-762","example-763","example-764","example-765","example-766","example-767","example-768","example-769","example-770","example-771","example-772","example-773","example-774","example-775","example-776","example-777","example-778","example-779","example-780","example-781","example-782","example-783","example-784","example-785","example-786","example-787","example-788","example-789","example-790","example-791","example-792","example-793","example-794","example-795","example-796","example-797","example-798","example-799","example-800","example-801","example-802","example-803","example-804","example-805","example-806","example-807","example-808","example-809","example-810","example-811","example-812","example-813","example-814","example-815","example-816","example-817","example-818","example-819","example-820","example-821","example-822","example-823","example-824","example-825","example-826","example-827","example-828","example-829","example-830","example-831","example-832","example-833","example-834","example-835","example-836","example-837","example-838","example-839","example-840","example-841","example-842","example-843","example-844","example-845","example-846","example-847","example-848","example-849","example-850","example-851","example-852","example-853","example-854","example-855","example-856","example-857","example-858","example-859","example-860","example-861","example-862","example-863","example-864","example-865","example-866","example-867","example-868","example-869","example-870","example-871","example-872","example-873","example-874","example-875","example-876","example-877","example-878","example-879","example-880","example-881","example-882","example-883","example-884","example-885","example-886","example-887","example-888","example-889","example-890","example-891","example-892","example-893","example-894","example-895","example-896","example-897","example-898","example-899","example-900","example-901","example-902","example-903","example-904","example-905","example-906","example-907","example-908","example-909","example-910","example-911","example-912","example-913","example-914","example-915","example-916","example-917","example-918","example-919","example-920","example-921","example-922","example-923","example-924","example-925","example-926","example-927","example-928","example-929","example-930","example-931","example-932","example-933","example-934","example-935","example-936","example-937","example-938","example-939","example-940","example-941","example-942","example-943","example-944","example-945","example-946","example-947","example-948","example-949","example-950","example-951","example-952","example-953","example-954","example-955","example-956","example-957","example-958","example-959","example-960","example-961","example-962","example-963","example-964","example-965","example-966","example-967","example-968","example-969","example-970","example-971","example-972","example-973","example-974","example-975","example-976","example-977","example-978","example-979","example-980","example-981","example-982","example-983","example-984","example-985","example-986","example-987","example-988","example-989","example-990","example-991","example-992","example-993","example-994","example-995","example-996","example-997","example-998","example-999"]
        
        Attribute items supports 1 item maximum, but config has 1000 declared.
--- FAIL: TestAccCloudflareTeamsList_LottaListItems (0.20s)
=== RUN   TestAccCloudflareTeamsList_Reordered
    resource_cloudflare_teams_list_test.go:129: Step 1/2 error: Error running pre-apply plan: exit status 1
        
        Error: Too many list items
        
          with cloudflare_teams_list.gtznntnfbp,
          on terraform_plugin_test.tf line 17, in resource "cloudflare_teams_list" "gtznntnfbp":
          17: 	items       = ["asdf-1234", "asdf-5678"]
        
        Attribute items supports 1 item maximum, but config has 2 declared.
--- FAIL: TestAccCloudflareTeamsList_Reordered (0.14s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	5.692s
FAIL
make: *** [testacc] Error 1
```

**Comment by rexscaria** [comment]
No the existing tests should work. Let me fix it.

File: internal/sdkv2provider/schema_cloudflare_teams_list.go (hunk 1 above)
Comment (rexscaria) [reply internal/sdkv2provider/schema_cloudflare_teams_list.go]: Just pushed a fix changing it. That was accidentally left out.

**Comment by rexscaria** [comment]
Test is fixed. i have squashed your commits @jacobbednarz you might need a force pull

**Comment by jacobbednarz** [comment]
looks good! thanks

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsList_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareTeamsList_Basic
--- PASS: TestAccCloudflareTeamsList_Basic (10.05s)
=== RUN   TestAccCloudflareTeamsList_BasicWithDescription
--- PASS: TestAccCloudflareTeamsList_BasicWithDescription (2.93s)
=== RUN   TestAccCloudflareTeamsList_LottaListItems
--- PASS: TestAccCloudflareTeamsList_LottaListItems (17.10s)
=== RUN   TestAccCloudflareTeamsList_Reordered
--- PASS: TestAccCloudflareTeamsList_Reordered (4.48s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	35.785s
```

**Pull request merged**